infer/environment.cc

```diff
@@ -1129,7 +1129,10 @@ Environment::processBinding(core::Context ctx, const cfg::CFG &inWhat, cfg::Bind
                 // by desugar and redacting the `SendResponse` so LSP features work
                 // more like developers expect.
                 const bool isDesugarTripleEqSend = send.fun == core::Names::tripleEq() && send.funLoc.empty();
-                if (lspQueryMatch && !isDesugarTripleEqSend) {
+                // Something like `X = Foo` will be rewritten to `X = Magic.<suggest-constant-type>(Foo)`
```

Comment:
My understanding is that this is only true if `Foo` failed to resolve:

https://github.com/sorbet/sorbet/blob/fd4f7df9f366f91976f9727a34fe5800227084de/resolver/resolver.cc#L2163-L2175

Can we rewrite the comment to make that clear?